staging/vhost-device-gpu/src/virt_gpu.rs

```diff
@@ -829,3 +830,68 @@ impl VirtioGpuBackend for VirtioGpu {
         Ok(OkNoData)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::sync::{Arc, Mutex};
+
+    use super::{VirtioGpu, VirtioGpuBackend, VirtioGpuResource, VirtioGpuRing, VirtioShmRegion};
+    use rutabaga_gfx::{
+        ResourceCreateBlob, RutabagaBuilder, RutabagaComponentType, RutabagaHandler,
+    };
+    use vhost_user_backend::{VringRwLock, VringT};
+    use vm_memory::{GuestAddress, GuestMemoryAtomic, GuestMemoryMmap};
+
+    fn new_2d() -> VirtioGpu {
+        let rutabaga = RutabagaBuilder::new(RutabagaComponentType::Rutabaga2D, 0)
+            .build(RutabagaHandler::new(|_| {}), None)
+            .unwrap();
+        VirtioGpu {
+            rutabaga,
+            resources: Default::default(),
+            fence_state: Arc::new(Mutex::new(Default::default())),
+        }
+    }
+
+    #[test]
+    fn test_gpu_backend_success() {
+        let mem = GuestMemoryAtomic::new(
```

Comment:
The `mem` and `vring` variables are not necessary for the test,  they can be removed.